wordpress-mobile/WordPress-Android
# Extract source interface for media picker

Fixes #12638 

This PR replaces the original DeviceListBuilder that was used to load media from the device with an interface and a class MediaLoader. This interface and the loader should support all kind of sources in the future. The loader supports pagination, refresh of data and error responses (and could be extended more). The functionality should be unchanged right now. 

The pagination is not used at the moment because the device list doesn't need to be paginated. We might add it as a feature in the future (because loading a lot of items might not be very efficient).

To test:
- test that the following functionality works as expected
- GB - Image block - Add image from device
- GB - Video block - Add video from device
- GB - Gallery block - Add images from device
- Add story -> Choose images and videos

PR submission checklist:
